pkg/deploy/controller/deploymentconfig/metrics.go

```diff
@@ -0,0 +1,22 @@
+package deploymentconfig
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// DeploymentConfigControllerSubsystem is how this controller is represented in
+// prometheus metrics.
+const DeploymentConfigControllerSubsystem = "deploymentconfig_controller"
+
+var (
+	deploymentConditionCounter = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: DeploymentConfigControllerSubsystem,
+			Name:      "set_condition",
+			Help:      "Gauge measuring changes to deployment conditions per condition, reason and the status",
```

Comment:
This metric is pretty vague. I would be interested in the ratio a deployment is flipping between True/False for Available, how many deployments are unavailable, how many deployments have failed over a period of time.